src/main/java/de/tum/in/www1/artemis/service/metis/ConversationService.java

```diff
@@ -0,0 +1,164 @@
+import de.tum.in.www1.artemis.repository.UserRepository;
+import de.tum.in.www1.artemis.repository.metis.ConversationParticipantRepository;
+import de.tum.in.www1.artemis.repository.metis.ConversationRepository;
+import de.tum.in.www1.artemis.security.Role;
+import de.tum.in.www1.artemis.service.AuthorizationCheckService;
+import de.tum.in.www1.artemis.web.rest.errors.BadRequestAlertException;
+import de.tum.in.www1.artemis.web.websocket.dto.metis.ConversationDTO;
+import de.tum.in.www1.artemis.web.websocket.dto.metis.CrudAction;
+
+@Service
+public class ConversationService {
+
+    private static final String CONVERSATION_ENTITY_NAME = "messages.conversation";
+
+    private static final String CONVERSATION_DETAILS_ENTITY_NAME = "messages.conversationParticipant";
+
+    private static final String METIS_WEBSOCKET_CHANNEL_PREFIX = "/topic/metis/";
+
+    private final UserRepository userRepository;
+
+    private final CourseRepository courseRepository;
+
+    private final AuthorizationCheckService authorizationCheckService;
+
+    private final ConversationParticipantRepository conversationParticipantRepository;
+
+    private final ConversationRepository conversationRepository;
+
+    private final SimpMessageSendingOperations messagingTemplate;
+
+    public ConversationService(UserRepository userRepository, CourseRepository courseRepository, AuthorizationCheckService authorizationCheckService,
+            ConversationParticipantRepository conversationParticipantRepository, ConversationRepository conversationRepository, SimpMessageSendingOperations messagingTemplate) {
+        this.userRepository = userRepository;
+        this.courseRepository = courseRepository;
+        this.authorizationCheckService = authorizationCheckService;
+        this.conversationParticipantRepository = conversationParticipantRepository;
+        this.conversationRepository = conversationRepository;
+        this.messagingTemplate = messagingTemplate;
+    }
+
+    /**
+     * Persists given conversation
+     *
+     * @param courseId      id of course the conversation belongs to
+     * @param conversation  conversation to be persisted
+     * @return              persisted conversation
+     */
+    public Conversation createConversation(Long courseId, Conversation conversation) {
+        final User user = this.userRepository.getUserWithGroupsAndAuthorities();
+
+        if (conversation.getId() != null) {
+            throw new BadRequestAlertException("A new conversation cannot already have an ID", CONVERSATION_ENTITY_NAME, "idexists");
+        }
+
+        if (conversation.getConversationParticipants().isEmpty()
+                || conversation.getConversationParticipants().stream().anyMatch(conversationParticipant -> conversationParticipant.getUser().getId().equals(user.getId()))) {
+            throw new BadRequestAlertException("A new conversation must have other conversationParticipants", CONVERSATION_DETAILS_ENTITY_NAME, "NotNull");
```

Comment:
```suggestion
            throw new BadRequestAlertException("A new conversation must have other conversation participants", CONVERSATION_DETAILS_ENTITY_NAME, "NotNull");
```
The error message should not display any implementation details.